pkg/comp-functions/functions/vshnredis/restore.go

```diff
@@ -0,0 +1,314 @@
+package vshnredis
+
+import (
+	"context"
+	_ "embed"
+	"strings"
+
+	vshnv1 "github.com/vshn/appcat/apis/vshn/v1"
+	runtime "github.com/vshn/appcat/pkg/comp-functions/runtime"
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	controllerruntime "sigs.k8s.io/controller-runtime"
+)
+
+//go:embed script/prepRestore.sh
+var prepRestoreScript string
+
+//go:embed script/restore.sh
+var restoreScript string
+
+//go:embed script/cleanupRestore.sh
+var cleanupRestoreScript string
+
+func RestoreBackup(ctx context.Context, iof *runtime.Runtime) runtime.Result {
+	log := controllerruntime.LoggerFrom(ctx)
+	log.Info("Starting RestoreBackup function")
+
+	comp := &vshnv1.VSHNRedis{}
+	err := iof.Observed.GetComposite(ctx, comp)
+	if err != nil {
+		return runtime.NewFatalErr(ctx, "Cannot get composite from function io", err)
+	}
+
+	// Wait for the next reconciliation in case instance namespace is missing
+	if comp.Status.InstanceNamespace == "" {
+		return runtime.NewWarning(ctx, "Composite is missing instance namespace, skipping transformation")
+	}
+
+	if comp.Spec.Parameters.Restore.BackupName == "" && comp.Spec.Parameters.Restore.ClaimName == "" {
+		return runtime.NewNormal()
+	}
+
+	if comp.Spec.Parameters.Restore.ClaimName == "" {
+		return runtime.NewWarning(ctx, "Composite is missing claimName parameter to restore from backup")
+	}
+
+	if comp.Spec.Parameters.Restore.BackupName == "" {
+		return runtime.NewWarning(ctx, "Composite is missing backupName parameter to restore from backup")
+	}
+
+	log.Info("Prepare for restore")
+
+	err = prepareRestoreJob(ctx, comp, iof)
+	if err != nil {
+		return runtime.NewFatalErr(ctx, "Can't deploy restore Job", err)
+	}
+
+	log.Info("Restoring from backup")
+
+	err = deployRestoreJob(ctx, comp, iof)
+	if err != nil {
+		return runtime.NewFatalErr(ctx, "Can't restore from backup", err)
+	}
+
+	log.Info("Cleanup restore")
+
+	err = cleanUpJob(ctx, comp, iof)
+	if err != nil {
+		return runtime.NewFatalErr(ctx, "Can't cleanup after restore", err)
+	}
+
```

Comment:
Tiny Nitpick: Naming of these helper functions could be a bit better. Something like `addPrepareRestoreJob` etc.

Right now while reading this I thought: "Ok we're first preparing the restore job, then we're deploying the restore job and then were cleaning up the job", which didn't really make sense.

Also the error messages are a bit misleading too. It's not that we "Can't restore from backup", we can't deploy the restore job.